internal/resourcepolicies/resource_policies.go

```diff
@@ -0,0 +1,109 @@
+package resourcepolicies
+
+// Action defined as one action for a specific way of backup
+type Action struct {
+	// Type defined specific type of action, it could be 'file-system-backup', 'volume-snapshot', or 'skip' currently
+	Type string `yaml:"type"`
+	// Parameters defined map of parameters when executing a specific action
+	// +optional
+	// +nullable
+	Parameters map[string]interface{} `yaml:"parameters,omitempty"`
+}
+
+// VolumePolicy defined policy to conditions to match Volumes and related action to handle matched Volumes
+type VolumePolicy struct {
+	// Conditions defined list of conditions to match Volumes
+	Conditions map[string]interface{} `yaml:"conditions"`
+	Action     Action                 `yaml:"action"`
+}
+
+// ResourcePolicies currently defined slice of volume policies to handle backup
+type ResourcePolicies struct {
+	Version        string         `yaml:"version"`
+	VolumePolicies []VolumePolicy `yaml:"volumePolicies"`
+	// we may support other resource policies in the future, and they could be added separately
+	// OtherResourcePolicies: []OtherResourcePolicy
+}
+
+type ResourcePolicyMatcher interface {
+	Match(res interface{}, actionType string) bool
+}
+
+type VolumePolicyMatcherImpl struct {
+	policy      VolumePolicy
+	matcherFunc func(volume interface{}, actionType string) bool
+}
+
+func (vpm *VolumePolicyMatcherImpl) Match(volume interface{}, actionType string) bool {
+	return vpm.matcherFunc(volume, actionType)
+}
+
+// Match interface is implemented by VolumePolicy
+func (policy *VolumePolicy) Match(volume interface{}, actionType string) bool {
+	if policy.Action.Type != actionType {
+		return false
+	}
+
+	val, ok := volume.(*VolumeConditions)
```

Comment:
is the first param a condition or a resource?

Reply:
it's the resource, and it's also `VolumeConditions` type of volume.

Here all original resource volumes would be PVs or Pod Volumes, they have different structures.  In order to match volumes of all kinds, I introduced one VolumeConditions(Maybe I should change the name) type of volume. Before doing the match volumes action, volumes of all kinds should transfer all filed values to a new VolumeConditions volume, then the match volumes action will avoid to compatible with various kinds of volume.